x-pack/plugin/ml-feature-index-builder/src/main/java/org/elasticsearch/xpack/ml/featureindexbuilder/job/AggregationConfig.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.ml.featureindexbuilder.job;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.search.aggregations.AggregationBuilder;
+import org.elasticsearch.search.aggregations.AggregatorFactories;
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+/*
+ * Wrapper for the aggregations config part of a composite aggregation.
+ *
+ * For now just wraps aggregations from composite aggs.
+ *
+ */
+public class AggregationConfig implements Writeable, ToXContentObject {
+
+    private final AggregatorFactories.Builder aggregatorFactoryBuilder;
+
+    public AggregationConfig(AggregatorFactories.Builder aggregatorFactoryBuilder) {
+        this.aggregatorFactoryBuilder = aggregatorFactoryBuilder;
+    }
+
+    AggregationConfig(final StreamInput in) throws IOException {
+        aggregatorFactoryBuilder = in.readOptionalWriteable(AggregatorFactories.Builder::new);
```

Comment:
It seems like there are unnecessarily many levels where `null` is allowed.  You're allowing `aggregatorFactoryBuilder` to be `null` here, but also in `FeatureIndexBuilderJobConfig` `aggregationConfig` is allowed to be `null`.  I think at most one of these possibilities should be allowed.